### PR TITLE
Pizza form

### PR DESCRIPTION
**Pull request by youngj3**
-new page for pizza form
-new post router to handle new pizza place addition 
-added route to render in front end App component
-updated client router

### CODE REVIEW

File: client/src/components/NewPizzaPlaceForm.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import React, { useState } from 'react'
+import { Redirect } from 'react-router-dom'
+import translateServerErrors from '../services/translateServerErrors.js'
+import ErrorList from '../components/layout/ErrorList.js'
```
Comment (nwalberts) [inline client/src/components/NewPizzaPlaceForm.js:4]: lets add an empty line after line 4

File: client/src/components/NewPizzaPlaceForm.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    const postNewPizzaPlace = async(newPizzaPlaceData) => {
+        try {
+            const response = await fetch("/api/v1/pizza-places", {
+                method: "POST",
+                headers: new Headers({
+                    "Content-Type": "application/json"
+                }),
+                body: JSON.stringify(newPizzaPlaceData)
+            })
+            console.log(response)
```
Comment (nwalberts) [inline client/src/components/NewPizzaPlaceForm.js]: no generic console.logs

File: client/src/components/NewPizzaPlaceForm.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                    const newErrors = translateServerErrors(body.errors)
+                    return setErrors(newErrors)
+                }else{
+                    throw new Error(`${response.status} (${response.statusText})`)
+                }
+            } else {
+                const body = await response.json()
+                              
+                setShouldRedirect(true)
+
```
Comment (nwalberts) [inline client/src/components/NewPizzaPlaceForm.js]: remove this line

File: client/src/components/NewPizzaPlaceForm.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                              
+                setShouldRedirect(true)
+               
+            }
+        } catch(error) {
+            console.error(`Error in fetch: ${error.message}`)
+        }
+    }
+
+
```
Comment (nwalberts) [inline client/src/components/NewPizzaPlaceForm.js]: remove extra line

File: client/src/components/NewPizzaPlaceForm.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+            name: "",
+            address: "",
+            phoneNumber: "",
+            website: "",
+            hours: "",
+            imageUrl: ""
+        })
+    }
+
+    if(shouldRedirect) {
```
Comment (nwalberts) [inline client/src/components/NewPizzaPlaceForm.js]: add a space after `if`

File: client/src/components/NewPizzaPlaceForm.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                name='imageUrl'
+                onChange={handleInputChange} 
+                value={newPizzaPlace.imageUrl}
+            />
+        </label>
+
+        <div>
+            <input className='button' type='submit' value='Submit' />
+            <input className='button' type='button' value='Clear Form' onClick={clearForm}/>
+        </div>
```
Comment (nwalberts) [inline client/src/components/NewPizzaPlaceForm.js]: this div doesnt do anything. Only add divs if you need a group of elements to style. remove it!

**Comment by JasperGroner** [comment]
You're probably going to want to get rid of the console.log in NewPizzaForm at line 28. Assume that was there for error checking.

Not something I personally care about, but the EEs don't like extra empty lines. Probably want to delete extra empty lines in NewPizzaForm at line 39, 41, 47

Likewise tabbing on return (line 77) of NewPizzaForm seems off.

File: client/src/components/PizzaPlaceList.js [hunk 7]
```diff
@@ -31,6 +32,7 @@ const PizzaPlaceList = (props) => {
         <div className='pizza-list'>
             <h1>Pizza Place List</h1>
             <ul>{pizzaPlaceReact}</ul>
+            <Link to='/pizza-places/new' >Add New Pizza Place</Link>
```
Comment (nwalberts) [inline client/src/components/PizzaPlaceList.js]: inconsistent spacing. Remove the extra space just after your `to` prop
I also recommend adding a space around `Add New Pizza Place`

File: server/src/routes/api/v1/pizzaPlaceRouter.js [hunk 8]
```diff
@@ -14,6 +15,23 @@ pizzaPlaceRouter.get("/", async (req, res) => {
     }
 })
 
+pizzaPlaceRouter.post("/", async (req, res) => {
+    const { body } = req
+    const formInput = cleanUserInput(body)
+
+    try{
```
Comment (nwalberts) [inline server/src/routes/api/v1/pizzaPlaceRouter.js]: add a space after `try`

File: server/src/routes/api/v1/pizzaPlaceRouter.js [hunk 9]
```diff
@@ -14,6 +15,23 @@ pizzaPlaceRouter.get("/", async (req, res) => {
...
+        const newPizzaPlace = await PizzaPlace.query().insertAndFetch(formInput)
+        return res.status(201).json({ newPizzaPlace })
+    } catch(error) {
+        if(error instanceof ValidationError) {
+            return res.status(422).json({
+                errors: error.data
+            })
```
Comment (nwalberts) [inline server/src/routes/api/v1/pizzaPlaceRouter.js]: this return doesnt need to be multiline. do it all on one lin!

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**